restful.entity.inc

```diff
@@ -35,18 +35,47 @@ function restful_entity_info() {
  * Implements hook_entity_update().
  */
 function restful_entity_update($entity, $type) {
-  list($entity_id) = entity_extract_ids($type, $entity);
-  $cid = 'paet:' . $type . '::ei:' . $entity_id;
-  \RestfulManager::invalidateEntityCache($cid);
+  if (restful_entity_determine_invalidation_by_type($type)) {
+    list($entity_id) = entity_extract_ids($type, $entity);
+    $cid = 'paet:' . $type . '::ei:' . $entity_id;
+    \RestfulManager::invalidateEntityCache($cid);
+  }
 }
 
 /**
  * Implements hook_entity_delete().
  */
 function restful_entity_delete($entity, $type) {
-  list($entity_id) = entity_extract_ids($type, $entity);
-  $cid = 'paet:' . $type . '::ei:' . $entity_id;
-  \RestfulManager::invalidateEntityCache($cid);
+  if (restful_entity_determine_invalidation_by_type($type)) {
+    list($entity_id) = entity_extract_ids($type, $entity);
+    $cid = 'paet:' . $type . '::ei:' . $entity_id;
+    \RestfulManager::invalidateEntityCache($cid);
+  }
+}
+
+/**
+ * Helper function, used in the entity hooks.
+ * 
+ * To determine whether entities are supposed to be cached or not.
+ * 
+ * @param string $entity_type
+ *   Entity type to be tested.
+ * @return bool 
+ */
+
+function restful_entity_determine_invalidation_by_type($entity_type) {
+  $skip_types = array(
+    // Saves to the configuration entity don't necessarily need to be
+    // cached & invalidated.   
+    'rate_limit',
+    // User is being dealt with separately.
+    'user'
+  );
+  $skip_types = array_merge(
+    $skip_types, 
+    variable_get('restful_entities_to_skip', array())
```

Comment:
I feel that if this `'restful_entities_to_skip'` is not documented, it will not be used. We should probably add a line to the documentation.

After that this PR is ready to be merged.